### PR TITLE
Add GitHub action for testing

### PR DESCRIPTION
**Pull request by zmbc**
~Currently failing because of the issues addressed in #6 (and #5).~

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        run: |
+          pip install -r pip.lock.txt
+          pip install .
+      - name: Run pipeline
+        run: |
+          snakemake --forceall
+      - name: Install lint dependencies
+        run: |
+          pip install .[dev]
+      - name: Lint
```
Comment (patricktnast) [inline .github/workflows/test.yaml]: I'm guessing you're doing the linting after the run itself to test the generated notebooks?

**Review by patricktnast — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (zmbc) [reply .github/workflows/test.yaml]: Not really -- I was thinking I didn't want to install the dev dependencies before doing the run. I should likely just make this two separate jobs.

**Comment by zmbc** [comment]
@patricktnast I've updated this pretty substantially, want to take another look? (sorry for all the commits, not easy to test this locally)

**Review by patricktnast — APPROVED** [review]
To summarize the main points since I last reviewed, you split the lint check from the tests, using a conda environment in the tests, and a few linting fixes?

**Comment by zmbc** [comment]
Yes!

**Pull request merged**